utility/SPIDEV/gpio.cpp

```diff
@@ -27,152 +32,83 @@ GPIO::~GPIO()
 {
 }
 
-void GPIO::open(int port, int DDR)
-{
-    FILE* f;
-    f = fopen("/sys/class/gpio/export", "w");
-    if (f == NULL) {
-        throw GPIOException("can't export GPIO pin .check access rights");
-    }
-    fprintf(f, "%d\n", port);
-    fclose(f);
-
-    int counter = 0;
-    char file[128];
-    sprintf(file, "/sys/class/gpio/gpio%d/direction", port);
-
-    while ((f = fopen(file, "w")) == NULL) { //Wait 10 seconds for the file to be accessible if not open on first attempt
-        sleep(1);
-        counter++;
-        if (counter > 10) {
-            throw GPIOException("can't access /sys/class/gpio/gpio%d/direction GPIO pin. check access rights");
-            /*perror("Could not open /sys/class/gpio/gpio%d/direction");
-            exit(0); */
-        }
-    }
-    int l = (DDR == 0) ? fprintf(f, "in\n") : fprintf(f, "out\n");
-    if (!(l == 3 || l == 4)) {
-        fclose(f);
-        throw GPIOException("can't set direction on GPIO pin. check access rights");
-    }
-    /*
-    if (DDR == 0)
-        fprintf(f, "in\n");
-    else
-        printf(f, "out\n");
-    */
-    fclose(f);
+int fd;
```

Comment:
We should probably initialize this to `-1` just to be safe. Upon further reading, I don't think this is needed. All GPIO functions (except `GPIO::open()`) use their own scoped variable `fd`.

Reply:
Any objections to removing this and instead using `int fd` in `GPIO::open()`?

Reply:
Nope, looks like that should be in `GPIO::open()` I'll make the change